packages/core/admin/admin/src/content-manager/utils/getMaxTempKey.js

```diff
@@ -3,9 +3,13 @@ const getMaxTempKey = (arr) => {
     return -1;
   }
 
+  /**
+   * We check if there are ids or __temp_key__ because you may add an object to a list of objects
+   * already in the DB.
+   */
   const maxTempKey = Math.max.apply(
     Math,
-    arr.map((o) => o.__temp_key__ ?? 0)
+    arr.map((o) => o.id ?? o.__temp_key__ ?? 0)
```

Comment:
I don't understand the issue you're solving here or why we need to adjust the temp key in this way, could you re-explain please?

Reply:
Say you have list of components in a dynamic zone with the following id's (the __temp_keys__ are set based on the index upon loading the data):

```
[
   {id: 2, __temp_key__: 0, __component: 'blog.simple' ....},
   {id: 3, __temp_key__: 1, __component: 'blog.simple'....}
]
```

Without the fix, when adding a new item to this zone, the `maxTempKey` function is called to retrieve a temporary client side id. This id, retrieved based on the highest `__temp_key__` in the list is eventually also set as id in the list:
```
[
   {id: 2, __temp_key__: 0, __component: 'blog.simple' ....},
   {id: 3, __temp_key__: 1, __component: 'blog.simple'....}
   {id: 2, __temp_key__: 2, __component: 'blog.simple'....}
]
```
This results in 2 items having the same id, which leads to weird, undefined behaviour.

The fix takes the `id` into account within `maxTempKey`, which in this case would result in:
```
[
   {id: 2, __temp_key__: 0, __component: 'blog.simple' ....},
   {id: 3, __temp_key__: 1, __component: 'blog.simple'....}
   {id: 4, __temp_key__: 4, __component: 'blog.simple'....}
]
```

Note that I didn't come up with this fix myself, it is copied from https://github.com/strapi/strapi/blob/d999ad0b00ff427885462051cdc96626b7946472/packages/core/admin/ee/admin/src/pages/SettingsPage/pages/ReviewWorkflows/reducer/index.js#L209-L214 as I assume the same key conflicting issue was present within the stages of review workflows.

---

Hop this clears things up :)